ktoml-core/src/commonMain/kotlin/com/akuleshov7/ktoml/decoders/TomlAbstractDecoder.kt

```diff
@@ -20,12 +19,46 @@ public abstract class TomlAbstractDecoder : AbstractDecoder() {
     private val localDateTimeSerializer = LocalDateTime.serializer()
     private val localDateSerializer = LocalDate.serializer()
 
-    // Invalid Toml primitive types, we will simply throw an error for them
-    override fun decodeByte(): Byte = invalidType("Byte", "Long")
-    override fun decodeShort(): Short = invalidType("Short", "Long")
-    override fun decodeInt(): Int = invalidType("Int", "Long")
-    override fun decodeFloat(): Float = invalidType("Float", "Double")
-    override fun decodeChar(): Char = invalidType("Char", "String")
+    override fun decodeByte(): Byte = decodePrimitiveIntegerType { it.toByte() }
+
+    override fun decodeShort(): Short = decodePrimitiveIntegerType { it.toShort() }
+
+    override fun decodeInt(): Int = decodePrimitiveIntegerType { it.toInt() }
+
+    override fun decodeFloat(): Float {
+        val keyValue = decodeKeyValue()
+
+        val doubleValue = keyValue.castOrThrow<Double>()
+
+        val floatValue = doubleValue.toFloat()
+
+        if (floatValue.isInfinite()) {
```

Comment:
Json apparently doesn't have any way to express infinity and NaN, so I guess that's why KxS has this exception. Toml does have `inf` and `nan` though

Reply:
Looks like I actually missed that, need to support it...